src/components/AllItemsHeader/AllItemsHeader.jsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from "react";
+import { useNavigate } from "react-router";
+import styles from "./AllItemsHeader.module.scss";
+
+export default function AllItemsHeader({ sort, setSort, setPage }) {
+  const navigate = useNavigate();
+  const [open, setOpen] = useState(false);
+
+  //정렬 방식이 바뀌면, 1번 페이지로 돌아감
+  function sortClick(sortIn) {
+    if (sortIn === "recent" && sort !== "recent") {
+      setSort("recent");
+      setPage(1);
+    } else if (sortIn === "favorite" && sort !== "favorite") {
+      setSort("favorite");
+      setPage(1);
+    }
+  }
```

Comment:
💊 제안
로직이 불필요하게 복잡한 것 같아요~ sortIn 즉 선택된 정렬값은 "recent" | "favorite" 이니 아래처럼 작성하셔도 충분합니다.

```suggestion
    if(sortIn === sort) return; // 기존과 같은 값이 선택될 경우 1페이지로 가는 것을 막기위해

    setSort(sortIn);
    setPage(1);
  }
```